golang/README.md

```diff
@@ -1,15 +1,65 @@
 # Golang at Clever
 
+## Onboarding
+
+It is suggested that you introduce yourself to Golang by first completing [the tour](https://tour.golang.org/welcome/1).
+Most topics are covered to help you hit the ground running.
+
+Most Go idioms can then be learned by reading [Effective Go]Effective Go](https://golang.org/doc/effective_go.html).
+This is highly suggested before you begin writing Go code for production.
+
+[The Go Blog](http://blog.golang.org/) contains many useful articles going over how to use several essential Go features like go-routines, JSON, constants, etc.
+
+
 ## Golang versions
 
-All golang services (with some exceptions, like Heka) at Clever should be on go-1.5.
+All golang services written by Clever should be on Go 1.5.
+Some 3rd-party tools that we use, such as Heka, are not on Go 1.5.
+
+
+## Best Practices
+
+### Sets
+
+Sets are typically best represented by a `map[key type]bool` structure where every boolean value is set to `true`.
```

Comment:
I've seen this library before: https://github.com/fatih/set

IIRC, it uses an underlying `map[interface{}]struct{}` as well, but also provides a nice abstraction overall if we specifically need a set structure for a larger use case.